Updated puppet-module-autofs

kividiot opened the pull request:
Major overhaul of the module. With checks, specs and whistles.

Commit:
Updated several aspects of autofs puppet module

- Added checks for supported OS
- Added check for values
- Added templates for config and auto.master files
- Added spec tests

Fixed formatting on README

Updated README examples

Updated supported OS versions

Fixed template code

Added switch for UMOUNT_WAIT

Added better validation of values and fixed bools in spec tests

Added support for external data for package_name and service_name

Added support for external data for package_name and service_name

Fix smaller code issues, update tests and new travis config

- Updated .gitignore
- Added metadata.json
- Updated README.md with new names for variables
- Updated init.pp with new code and names for variables
- Updated init_spec.rb to test new variables
- Updated templates to use new names for variables

Updated travis file

Added Gemfile and fix for Travis file

Updated versions in metadata.json and README.rd

Updated list of supported OSes

Commit:
Updated several aspects of autofs puppet module

- Added checks for supported OS
- Added check for values
- Added templates for config and auto.master files
- Added spec tests

Commit:
Fixed formatting on README

Commit:
Updated README examples

Commit:
Updated supported OS versions

Commit:
Fixed template code

Commit:
Added switch for UMOUNT_WAIT

Commit:
Added better validation of values and fixed bools in spec tests

kividiot commented:
Could you have a look at this?

Commit:
Added support for external data for package_name and service_name

Commit:
Added support for external data for package_name and service_name

ghoneycutt commented:
Awesome work!

ghoneycutt commented on Modulefile:
```diff
@@ -7,4 +7,4 @@ summary 'autofs'
 description 'Manage automounter'
 project_page 'https://github.com/ghoneycutt/puppet-module-autofs'
 
-dependency 'ghoneycutt/nfs', '>= 0.0.1'
```
Why no longer the dependency on NFS? I'm ok with that, though this is a breaking change, which is also OK. Just want to understand the why.

kividiot replied:
While automounts are mostly nfs the puppet modules is not needed.

ghoneycutt replied:
right on

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
+
+This module has been tested to work on the following systems with Puppet v3 and Ruby versions 1.8.7, 1.9.3 and 2.0.0.
+
+  * RedHat EL5 & EL6
```
Listed as

```
* Debian 7
* EL 5
* EL 6
* Suse 10
* Suse 11
* Ubuntu 12.04
```

ghoneycutt replied:
metadata.json also needs this info

kividiot replied:
Done

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,29 +2,90 @@
+  $browsable        = 'no',
+  $mounts           = undef,
+  $package_name     = 'autofs',
+  $service_name     = 'autofs',
```
align arrows one space after longest parameter

kividiot replied:
Done

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,29 +2,90 @@
+  $mounts           = undef,
+  $package_name     = 'autofs',
+  $service_name     = 'autofs',
+){
```
space between `)` and `{`

kividiot replied:
Done

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,29 +2,90 @@
 #
```
`# == Class: autofs`

ghoneycutt commented on templates/auto_master.erb:
```diff
@@ -0,0 +1,8 @@
+#
```
Use standard header

https://github.com/ghoneycutt/puppet-module-dnsclient/blob/master/templates/resolv.conf.erb#L1-2

kividiot replied:
Done

ghoneycutt commented on templates/auto_master.erb:
```diff
@@ -0,0 +1,8 @@
+#
+
+<% if @mounts -%>
+<% @mounts.each do | mount| -%>
```
Is an array ok here or do we need a hash? I'm not familiar enough to know if you need to specify per mount options or if that is even a possibility.

ghoneycutt replied:
The array should be sorted, else it could come out different between puppet runs with ruby 1.8.7 and cause churn.

kividiot replied:
It's an Array with each row in auto.master as an array value.

sort added

ghoneycutt commented on templates/debian-default-autofs.erb:
```diff
@@ -0,0 +1,145 @@
+#
```
Use standard header

https://github.com/ghoneycutt/puppet-module-dnsclient/blob/master/templates/resolv.conf.erb#L1-2

ghoneycutt replied:
This template looks almost the same as for EL. It would be better to use one template instead of two.

kividiot replied:
Done

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,29 +2,90 @@
+      require => Package[$package_name],
+    }
+
+  } else {
```
service should not be listed twice. Instead have the if enabled == true block set a variable for service_ensure and service_enable and use those to either enable or disable the service.

ghoneycutt commented:
For travis, suggest grabbing Gemfile, travis.yml, Rakefile and spec/spec_helper.rb from my vim module.

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
-Automounter for NFS.
+# Compatibility
+
+This module has been tested to work on the following systems with Puppet v3 and Ruby versions 1.8.7, 1.9.3 and 2.0.0.
```
With the new travis.yml, this will also support ruby 2.1.0 Puppet v3 with the future parser and Puppet v4

kividiot replied:
Done

ghoneycutt replied:
Suggest changing to

`This module has been tested to work on the following systems with Puppet v3 (with and without the future parser) and Puppet v4 with Ruby versions 1.8.7, 1.9.3, 2.0.0 and 2.1.0.`

kividiot replied:
Done

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
+
+- *Default*: true
+
+mounttimeout
```
`mount_timeout` is a bit easier to read.

kividiot replied:
Done

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
+
+- *Default*: 300
+
+umountwait
```
`unmount_wait`

kividiot replied:
Done

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
+
+umountwait
+----------
+Numeric value to try and prevent expire delays when trying to umount from a server
```
What happens if this is undef as opposed to a number? Can this be 0? Can it be negative? If it needs to be positive or non-zero, please note that and add an additional check/test.

kividiot replied:
Changed mount_timeout and umount_wait checks to validate_numeric with a lowest value of 1.

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
+
+package_name
+------------
+Name of package to install for automount functionality
```
`String to specify the name of the package to install for automount.`

kividiot replied:
Done

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
+
+mounts
+------
+Array value to set mount paths.
```
`Array of mount paths.`

ghoneycutt replied:
in stdlib 4.6.0 `validate_absolute_path()` now supports arrays, so use that to validate this. I believe the mounts have to be absolute paths right? If so, that should also be mentioned in the README.

ghoneycutt replied:
Nevermind, i see that the array can have a lot of different things. In which case, just use `validate_array()`

kividiot replied:
Done

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
+
+service_name
+------------
+Name of service to handle for automount functionality
```
`String to specify name of the autofs service.`

kividiot replied:
Done

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,73 @@
-If the file 'auto.master' changes, the service 'autofs' is notifed and restarted.
-The service 'autofs' depends on the 'idmapd_service' service.
+<pre>
+autofs::enable: true
```
This is a default value, so we don't need it here.

kividiot replied:
Made a non default example

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,29 +2,90 @@
+        owner   => 'root',
+        group   => 'root',
+        mode    => '0644',
+        content => $::osfamily ? {
```
Use a single template and remove this. We actually do not want to ever use the selector inside of a resource.

kividiot replied:
The templates for Debian/Ubuntu, RedHat and Suse are not exhanceable. Moved the template selection to other OS related ifs

Commit:
Fix smaller code issues, update tests and new travis config

- Updated .gitignore
- Added metadata.json
- Updated README.md with new names for variables
- Updated init.pp with new code and names for variables
- Updated init_spec.rb to test new variables
- Updated templates to use new names for variables

Commit:
Updated travis file

Commit:
Added Gemfile and fix for Travis file

ghoneycutt commented on metadata.json:
```diff
@@ -0,0 +1,77 @@
+  "requirements": [
+    {
+      "name": "pe",
+      "version_requirement": ">= 3.2.0 < 4.0.0"
```
`"version_requirement": ">= 3.1.0 < 5.0.0"`

kividiot replied:
Done

ghoneycutt commented on metadata.json:
```diff
@@ -0,0 +1,77 @@
+    },
+    {
+      "name": "puppet",
+      "version_requirement": ">= 3.0.0 < 4.0.0"
```
`"version_requirement": ">= 3.1.0 < 5.0.0"`

kividiot replied:
Done

Commit:
Updated versions in metadata.json and README.rd

ghoneycutt commented on README.md:
```diff
@@ -1,9 +1,74 @@
+
+This module has been tested to work on the following systems with Puppet v3 (with and without the future parser) and Puppet v4 with Ruby versions 1.8.7, 1.9.3, 2.0.0 and 2.1.0.
+
+  * RedHat EL5 & EL6
```
Suggest

```
* Debian 7
* EL5
* EL6
* Suse 10
* Suse 11
* Ubuntu 12.04
```

ghoneycutt commented:
Could you squash your commits when you are done and I'll get this merged

Commit:
Updated list of supported OSes

Commit:
Merge branch 'master' of github.com:kividiot/puppet-module-autofs
